src/lib.rs

```diff
@@ -45,7 +62,7 @@ pub fn get() -> Result<Duration, String> {
             .map(|d| d - Duration::new(boottime.tv_sec as u64, boottime.tv_usec as u32 * 1000))
             .map_err(|e| e.to_string())
```

Comment:
For the BSDs you also need to handle the result of [`SystemTime::duration_since`](https://doc.rust-lang.org/stable/std/time/struct.SystemTime.html#method.duration_since) (e.g. via a variant encapsulating an `std::time::SystemTimeError`)

Reply:
Very good catch! Thanks!